### PR TITLE
chore: refactor setting languages

### PR DESCRIPTION
**Pull request by UncleSamtoshi**
- add thai language support
- improve typing around languages
- fix bug with setting languages

### CODE REVIEW

File: app/utils/locale-detector.ts [hunk 1]
```diff
@@ -1,38 +1,60 @@
...
-    RNLocalize.getLocales() &&
-    RNLocalize.getLocales().some(
-      (locale) =>
-        locale.languageCode?.startsWith("en") ||
-        locale.languageCode?.startsWith("es") ||
-        locale.languageCode?.startsWith("fr") ||
-        locale.languageCode?.startsWith("pt") ||
-        locale.languageCode?.startsWith("de") ||
-        locale.languageCode?.startsWith("cs"),
+export const detectDefaultLocale = (): Locales => {
```
Comment (nicolasburtey) [inline app/utils/locale-detector.ts:5]: some tests on this could be useful

Comment (UncleSamtoshi) [reply app/utils/locale-detector.ts]: Added some tests

File: __tests__/utils/locale-detector.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { matchOsLocaleToSupportedLocale } from "../../app/utils/locale-detector"
```
Comment (nicolasburtey) [inline __tests__/utils/locale-detector.test.ts:1]: side note: I have been confused as some test are "in line", ie in the same folder as the other one, and other tests are in `__tests__`.

I dpn't really care which way we go, but having consistency here would be useful. feel free to take this PR if you'd like probably a small one

File: test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const test = function(a, b) {
```
[the remark this answers is not in the document]
Comment (UncleSamtoshi) [reply test.js:1]: Yes i will remove that

File: __tests__/utils/locale-detector.test.ts (hunk 2 above)
Comment (UncleSamtoshi) [reply __tests__/utils/locale-detector.test.ts:1]: I will create a follow up PR to bring consistency to the testing.

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**